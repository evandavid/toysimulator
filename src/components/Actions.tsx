import React from 'react';
import {View, TouchableWithoutFeedback} from 'react-native';
import {Row, AsButton, ButtonText} from '../styled';

type ActionsProps = {
  onAddPlace: () => void;
  onAddMove: () => void;
  onAddLeft: () => void;
  onAddRight: () => void;
  onAddReport: () => void;
  validForOtherAction: boolean;
};

const Actions = ({
  onAddPlace,
  onAddMove,
  onAddLeft,
  onAddRight,
  onAddReport,
  validForOtherAction,
}: ActionsProps) => {
  return (
    <View>
      <Row>
        <TouchableWithoutFeedback onPress={onAddPlace}>
          <AsButton>
            <ButtonText>PLACE</ButtonText>
          </AsButton>
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback
          onPress={onAddMove}
          disabled={!validForOtherAction}>
          <AsButton disabled={!validForOtherAction}>
            <ButtonText>MOVE</ButtonText>
          </AsButton>
        </TouchableWithoutFeedback>
      </Row>
      <Row>
        <TouchableWithoutFeedback
          onPress={onAddLeft}
          disabled={!validForOtherAction}>
          <AsButton disabled={!validForOtherAction}>
            <ButtonText>LEFT</ButtonText>
          </AsButton>
        </TouchableWithoutFeedback>
        <TouchableWithoutFeedback
          onPress={onAddRight}
          disabled={!validForOtherAction}>
          <AsButton disabled={!validForOtherAction}>
            <ButtonText>RIGHT</ButtonText>
          </AsButton>
        </TouchableWithoutFeedback>
      </Row>
      <TouchableWithoutFeedback
        onPress={onAddReport}
        disabled={!validForOtherAction}>
        <AsButton disabled={!validForOtherAction}>
          <ButtonText>REPORT</ButtonText>
        </AsButton>
      </TouchableWithoutFeedback>
    </View>
  );
};

export default Actions;
