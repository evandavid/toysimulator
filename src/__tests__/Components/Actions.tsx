import React from 'react';
import renderer from 'react-test-renderer';
import Actions from '../../components/Actions';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Actions
        onAddMove={jest.fn}
        onAddLeft={jest.fn}
        onAddRight={jest.fn}
        onAddReport={jest.fn}
        onAddPlace={jest.fn}
        onClear={jest.fn}
        validForOtherAction={false}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();

  const tree2 = renderer
    .create(
      <Actions
        onAddMove={jest.fn}
        onAddLeft={jest.fn}
        onAddRight={jest.fn}
        onAddReport={jest.fn}
        onAddPlace={jest.fn}
        onClear={jest.fn}
        validForOtherAction={true}
      />,
    )
    .toJSON();
  expect(tree2).toMatchSnapshot();
});
