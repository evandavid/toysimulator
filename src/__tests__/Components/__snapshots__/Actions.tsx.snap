// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "height": 105,
    }
  }
>
  <View
    className="sc-gsTEea jwzyjw"
  >
    <View
      accessible={true}
      className="sc-gKseQn bEKjdd"
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        className="sc-fubCzh GEpea"
      >
        PLACE
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      className="sc-gKseQn cDVuzZ"
      disabled={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        className="sc-fubCzh GEpea"
      >
        MOVE
      </Text>
    </View>
  </View>
  <View
    className="sc-gsTEea jwzyjw"
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      className="sc-gKseQn cDVuzZ"
      disabled={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        className="sc-fubCzh GEpea"
      >
        LEFT
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      className="sc-gKseQn cDVuzZ"
      disabled={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        className="sc-fubCzh GEpea"
      >
        RIGHT
      </Text>
    </View>
  </View>
  <View
    className="sc-gsTEea jwzyjw"
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      className="sc-gKseQn cDVuzZ"
      disabled={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        className="sc-fubCzh GEpea"
      >
        REPORT
      </Text>
    </View>
    <View
      accessible={true}
      className="sc-gKseQn bEKjdd"
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#3D5866",
        }
      }
    >
      <Text
        className="sc-fubCzh GEpea"
      >
        CLEAR
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly 2`] = `
<View
  style={
    Object {
      "height": 105,
    }
  }
>
  <View
    className="sc-gsTEea jwzyjw"
  >
    <View
      accessible={true}
      className="sc-gKseQn bEKjdd"
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        className="sc-fubCzh GEpea"
      >
        PLACE
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      className="sc-gKseQn bEKjdd"
      disabled={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        className="sc-fubCzh GEpea"
      >
        MOVE
      </Text>
    </View>
  </View>
  <View
    className="sc-gsTEea jwzyjw"
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      className="sc-gKseQn bEKjdd"
      disabled={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        className="sc-fubCzh GEpea"
      >
        LEFT
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      className="sc-gKseQn bEKjdd"
      disabled={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        className="sc-fubCzh GEpea"
      >
        RIGHT
      </Text>
    </View>
  </View>
  <View
    className="sc-gsTEea jwzyjw"
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      className="sc-gKseQn bEKjdd"
      disabled={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        className="sc-fubCzh GEpea"
      >
        REPORT
      </Text>
    </View>
    <View
      accessible={true}
      className="sc-gKseQn bEKjdd"
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#3D5866",
        }
      }
    >
      <Text
        className="sc-fubCzh GEpea"
      >
        CLEAR
      </Text>
    </View>
  </View>
</View>
`;
